### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by jamie-suse**
Adds integration tests to the TLint project

### CODE REVIEW

File: tests/cli.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+use std::process::Command;
+
+#[test]
+fn validates_check_file() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (jamie-suse) [inline tests/cli.rs:6]: How do you think we should go about adding an integration test for ensuring that TLint correctly invalidates an invalid Check file? Should I create an invalid Check inside `tests/`?

**Comment by jamie-suse** [comment]
Also if I'm missing some important tests please let me know!

File: tests/cli.rs (hunk 1 above)
Comment (dottorblaster) [reply tests/cli.rs:6]: Yes please :heart: feel free to move the `check.yml` file under `tests/fixtures` and create a generic invalid check to test against. Just one for the moment will do I think :+1:

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Great job! Onto the "failing check scenario" now, keep it up!

File: tests/fixtures/invalid_check.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Corosync configuration file
```
Comment (nelsonkopliku) [inline tests/fixtures/invalid_check.yml:1]: would you mind adding a commented line for the omitted id? I'd be clearer why this check is invalid

**Review by nelsonkopliku — APPROVED** [review]
LGTM :crab:

**Review by dottorblaster — APPROVED** [review]
LGTM

**Pull request merged**